src/interceptors/WebSocket/WebSocketTransport.ts

```diff
@@ -1,40 +1,39 @@
-export type WebSocketData = string | ArrayBufferLike | Blob | ArrayBufferView
-
-export type WebSocketTransportOnIncomingCallback = (
-  event: MessageEvent<WebSocketData>
-) => void
-
-export type WebSocketTransportOnOutgoingCallback = (data: WebSocketData) => void
+import { CloseEvent } from './utils/events'
 
-export type WebSocketTransportOnCloseCallback = (event: CloseEvent) => void
-
-export abstract class WebSocketTransport {
-  /**
-   * A callback for the incoming server events.
-   * This is called when the WebSocket client receives
-   * a message from the server.
-   */
-  abstract onIncoming: WebSocketTransportOnIncomingCallback
+export type WebSocketData = string | ArrayBufferLike | Blob | ArrayBufferView
 
-  /**
-   * A callback for outgoing client events.
-   * This is called when the WebSocket client sends data.
-   */
-  abstract onOutgoing: WebSocketTransportOnOutgoingCallback
+export type WebSocketTransportEventMap = {
```

Comment:
This pull request also includes an internal refactoring of the `WebSocketTransport`:

- No longer an abstract class but an interface. Implement it to your heart's content (which, in our case, is `EventTarget`).
- Transport is _event-based_ now since multiple agents can listen to the same events (incoming/outgoing).